src/Features/Core/Portable/Diagnostics/DocumentAnalysisExecutor.cs

```diff
@@ -187,6 +210,43 @@ private async Task<ImmutableArray<Diagnostic>> GetSyntaxDiagnosticsAsync(SyntaxT
                 ImmutableArray<Diagnostic>.Empty;
         }
 
+        private async Task<ImmutableArray<Diagnostic>> GetAdditionalDocumentDiagnosticsAsync(AdditionalDocument document, DiagnosticAnalyzer analyzer, bool isCompilerAnalyzer, CancellationToken cancellationToken)
+        {
+            // PERF: Compute diagnostics for all analyzers with a single invocation into CompilationWithAnalyzers.
```

Comment:
Its difficult for me to guess at how expensive this operation is. Do we have perf numbers for this?

Reply:
Actually, this is a requirement as part of https://github.com/dotnet/roslyn/pull/45607, where we move all IDE diagnostic analysis to OOP. So, even if perf improvements were not significant, we are going to batch the requests for multiple analyzers into a single one with all analyzers specified at once.